libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -828,6 +828,45 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				StorageByteArrayElement(m_context).storeValue(*type, _functionCall.location(), true);
 			break;
 		}
+		case FunctionType::Kind::ByteArrayPop:
+		case FunctionType::Kind::ArrayPop:
+		{
+			_functionCall.expression().accept(*this);
+			solAssert(function.parameterTypes().size() == 0, "");
+			// solAssert(!!function.parameterTypes()[0], "");
+			//TypePointer paramType = function.parameterTypes()[0];
+			shared_ptr<ArrayType> arrayType = make_shared<ArrayType>(DataLocation::Storage);
+				// function.kind() == FunctionType::Kind::ArrayPop ?
+				// make_shared<ArrayType>(DataLocation::Storage) :
+				// make_shared<ArrayType>(DataLocation::Storage);
+			// get the current length
+			ArrayUtils(m_context).retrieveLength(*arrayType);
+			m_context << Instruction::DUP1;
+			// stack: ArrayReference currentLength currentLength
+			m_context << u256(1) << Instruction::SUB;
+			// stack: ArrayReference currentLength newLength
+			m_context << Instruction::DUP3 << Instruction::DUP2;
+			ArrayUtils(m_context).resizeDynamicArray(*arrayType);
```

Comment:
We should probably write a new routine for this. I think `resizeDynamicArray` introduces a loop which is not necessary here. This might get a little complicated, but in general, you need to call:

```
StorageItem(_context, *_type).setToZero(SourceLocation(), false);
```

Also, I would not bother for ByteArrays for now, they are even more complicated :-)

Reply:
Why zero it in the first place? Just decrement the length field of the type that's all?

Reply:
@axic solidity always cleans storage.